docs/upgrade.html

```diff
@@ -63,6 +63,9 @@
     message format version. Otherwise consumers before 0.10.0.0 might break. In particular, after the message format
     is set to 0.10.0, one should not change it back to an earlier format as it may break consumers on versions before 0.10.0.0.
 </p>
+<p>
+    <b>Note:</b> Due to the additional timestamp introduced in each message, producers sending small messages may see a message throughput degrade because of the increased overhead.
+</p>
```

Comment:
Perhaps we can also add a note on the impact on batching. How about the following?

When receiving compressed messages, 0.10.0 brokers avoid recompressing the messages, which in general improves the throughput. In certain cases, this may reduce the batching size on the producer, which could lead to worse throughput. If this happens, users can tune linger.ms and batch.size in the producer for better throughput.